models/articles.model.js

```diff
@@ -37,9 +37,39 @@ exports.selectArticles = (sort_by = "created_at", order = "DESC") => {
   );
 
   return db.query(queryString).then(({ rows }) => {
-    // if (rows.length === 0) {
-    //   return Promise.reject({ status: 400, msg: "does not exist" });
-    // }
+ 
     return rows;
   });
 };
+
+exports.selectComments = (sort_by = "created_at", order = "DESC", article_id) => {
+  // Greenlist
+  const validOrders = ["ASC", "DESC"];
+  if (!validOrders.includes(order.toUpperCase())) {
+    return Promise.reject({ status: 400, msg: "Bad Request" });
+  }
+
+  const queryString = format(
+    `SELECT 
+      comments.author,
+      comments.body,
+      comments.article_id,
+      comments.created_at,
+      comments.votes,
+      comments.comment_id
+    FROM comments LEFT JOIN articles ON comments.article_id = articles.article_id
+    WHERE articles.article_id = %L
+    ORDER BY comments.%I %s;`,
+    article_id,
+    sort_by,
+    order
+   
+  );
+
+  return db.query(queryString).then(({ rows }) => {
+     if (rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Not found" });
```

Comment:
We need to determine why we are returning an empty array...
This could be bcause:
article exists but has no comments
OR
article does not exist.
Therefore, you will need to check whether the article exists in order to determine this.